fontembed/embed_pdf.c

```diff
@@ -125,6 +125,13 @@ static EMB_PDF_FONTDESCR *emb_pdf_fd_new(const char *fontname,
   ret->fontname=ret->data+len;
   len+=strlen(fontname)+1;
   if (subset_tag) {
+    //Allocate new array and copy in data
+    char *newArray = new char[7];
+    memcpy(newArray, ret->fontname,1);
+    //Delete old array
+    delete [] ret->fontname;
+    //Swap pointers and new size
+    ret->fontname = newArray;
     strncpy(ret->fontname,subset_tag,6);
     ret->fontname[6]='+';
     strcpy(ret->fontname+7,fontname);
```

Comment:
Hi @tillkamppeter 
I have tried fixing the issue, compiled and tested using (Ubuntu 9.3.0-17ubuntu1~20.04) is show no warnings, However since you mentioned you are getting the issue after updating to GCC 10.2.0, I tried updating my GCC sadly there is no possibly working the way I could find, not does the Ubuntu Groovy Gorilla works on Virtual Box or Live USB.
What do you recommend me to do now.
Thanks for the help.

Solution I am trying - Using snprintf ( char * s, size_t n, const char * format, ... ) in place of char *strncpy( char *dest, const char *src, size_t n )

Thanks in advance.